protocol/x/affiliates/keeper/keeper_test.go

```diff
@@ -14,3 +20,193 @@ func TestLogger(t *testing.T) {
 	logger := tApp.App.AffiliatesKeeper.Logger(ctx)
 	require.NotNil(t, logger)
 }
+
+func TestRegisterAffiliate(t *testing.T) {
+	tApp := testapp.NewTestAppBuilder(t).Build()
+	ctx := tApp.InitChain()
+	k := tApp.App.AffiliatesKeeper
+
+	tests := []struct {
+		name        string
+		referee     string
+		affiliate   string
+		expectError bool
+	}{
+		{
+			name:        "Register new affiliate",
+			referee:     "referee1",
```

Comment:
Change these into valid dYdX bech32 addresses; add additional test case for invalid address